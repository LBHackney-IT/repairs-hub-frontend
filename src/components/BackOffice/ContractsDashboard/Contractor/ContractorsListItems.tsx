import Contractor from '@/root/src/models/contractor'
import ContractorListItem from './ContractorListItem'
import WarningInfoBox from '../../../Template/WarningInfoBox'

interface ContractorsListItemsProps {
  contractors: Contractor[]
}

const ContractorsListItems = ({ contractors }: ContractorsListItemsProps) => {
  if (contractors.length === 0) {
    return (
      <>
        <div style={{ width: '85%' }}>
          <WarningInfoBox
            header="No contractors found!"
            text="Problem loading contractors."
            name="no-contractors-found"
          />
        </div>
      </>
    )
  }

  return (
    <div>
      <ol
        className="lbh-list mobile-working-work-order-list"
        data-test-id="contractors-list"
      >
        {contractors.map((obj, index) => (
          <ContractorListItem
            contractorReference={obj.contractorReference}
            contractorName={obj.contractorName}
            activeContractCount={obj.activeContractCount}
            key={index}
          />
        ))}
      </ol>
    </div>
  )
}

export default ContractorsListItems
