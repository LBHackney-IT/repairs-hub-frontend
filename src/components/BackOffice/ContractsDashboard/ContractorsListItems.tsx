import Contract from '@/root/src/models/contract'
import ContractorListItem from './ContractorListItem'
import WarningInfoBox from '../../Template/WarningInfoBox'

interface ContractorsListItemsProps {
  contracts: Contract[]
}

const ContractorsListItems = ({ contracts }: ContractorsListItemsProps) => {
  const relevantContracts = contracts?.filter((c) => c.terminationDate > '2020')

  const contractorNames = new Set<string>(
    relevantContracts.map((contract) => contract.contractorName)
  )
  const contractorReferences = new Set<string>(
    relevantContracts.map((contract) => contract.contractorReference)
  )

  const contractorNamesAndReferences = [...contractorNames].map(
    (name, index) => ({
      contractorName: name,
      contractorReference: [...contractorReferences][index],
    })
  )

  const contractorNamesAndReferencesSortedByContractorName = contractorNamesAndReferences.sort(
    (a, b) => a.contractorName.localeCompare(b.contractorName)
  )

  if (
    !contractorNamesAndReferencesSortedByContractorName ||
    Object.keys(contractorNamesAndReferencesSortedByContractorName).length === 0
  ) {
    return (
      <>
        <h3 className="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1">
          Contractors:
        </h3>
        <div style={{ width: '85%' }}>
          <WarningInfoBox
            header="No contractors found!"
            text="Problem loading contractors."
            name="no-contractors-found"
          />
        </div>
      </>
    )
  }

  return (
    <div>
      <h3 className="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1">
        Contractors:
      </h3>
      <ol className="lbh-list mobile-working-work-order-list">
        {contractorNamesAndReferencesSortedByContractorName.map(
          (obj, index) => (
            <ContractorListItem
              contractorReference={obj.contractorReference}
              contractorName={obj.contractorName}
              key={index}
            />
          )
        )}
      </ol>
    </div>
  )
}

export default ContractorsListItems
