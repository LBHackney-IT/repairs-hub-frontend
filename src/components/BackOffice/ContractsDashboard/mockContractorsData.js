export const mockContractors = [
  {
    contractorReference: 'NOO',
    contractorName: 'No ltd',
    activeContractCount: 0,
  },
  {
    contractorReference: 'YES',
    contractorName: 'Yes ltd',
    activeContractCount: 1,
  },
  {
    contractorReference: 'SHE',
    contractorName: 'She ltd',
    activeContractCount: 0,
  },
  {
    contractorReference: 'CAT',
    contractorName: 'Cat ltd',
    activeContractCount: 0,
  },
  {
    contractorReference: 'DOG',
    contractorName: 'Dog ltd',
    activeContractCount: 1,
  },
]
