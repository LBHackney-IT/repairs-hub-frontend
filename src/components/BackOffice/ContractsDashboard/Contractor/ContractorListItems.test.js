import { render } from '@testing-library/react'

import ContractorsListItems from './ContractorsListItems'

import { mockContractors } from '../mockContractorsData'

describe('Contractors list items component', () => {
  it('should render the component', async () => {
    const { asFragment } = render(
      <ContractorsListItems contractors={mockContractors} />
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
