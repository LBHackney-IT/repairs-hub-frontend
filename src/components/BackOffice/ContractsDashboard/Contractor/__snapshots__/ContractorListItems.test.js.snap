// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contractors list items component should render the component 1`] = `
<DocumentFragment>
  <div>
    <ol
      class="lbh-list mobile-working-work-order-list"
      data-test-id="contractors-list"
    >
      <li
        class="govuk-!-margin-top-3 operative-work-order-list-item"
        style="cursor: pointer; border: 5px solid #00664f; border-radius: 20px; width: 85%;"
      >
        <div
          class="contract-details"
        >
          <h3
            class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
          >
            HH Communal/Estate Inspes
          </h3>
          <p>
            Active contracts: 
            <span
              style="font-weight: 800;"
            />
          </p>
        </div>
        <div
          class="govuk-!-margin-0"
        >
          <span
            class="arrow right"
          />
        </div>
      </li>
      <li
        class="govuk-!-margin-top-3 operative-work-order-list-item"
        style="cursor: pointer; border: 5px solid #00664f; border-radius: 20px; width: 85%;"
      >
        <div
          class="contract-details"
        >
          <h3
            class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
          >
            HH Sign Services
          </h3>
          <p>
            Active contracts: 
            <span
              style="font-weight: 800;"
            />
          </p>
        </div>
        <div
          class="govuk-!-margin-0"
        >
          <span
            class="arrow right"
          />
        </div>
      </li>
      <li
        class="govuk-!-margin-top-3 operative-work-order-list-item"
        style="cursor: pointer; border: 5px solid #00664f; border-radius: 20px; width: 85%;"
      >
        <div
          class="contract-details"
        >
          <h3
            class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
          >
            Sycous Limited
          </h3>
          <p>
            Active contracts: 
            <span
              style="font-weight: 800;"
            />
          </p>
        </div>
        <div
          class="govuk-!-margin-0"
        >
          <span
            class="arrow right"
          />
        </div>
      </li>
      <li
        class="govuk-!-margin-top-3 operative-work-order-list-item"
        style="cursor: pointer; border: 5px solid #00664f; border-radius: 20px; width: 85%;"
      >
        <div
          class="contract-details"
        >
          <h3
            class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
          >
            Herts Heritage Ltd
          </h3>
          <p>
            Active contracts: 
            <span
              style="font-weight: 800;"
            />
          </p>
        </div>
        <div
          class="govuk-!-margin-0"
        >
          <span
            class="arrow right"
          />
        </div>
      </li>
      <li
        class="govuk-!-margin-top-3 operative-work-order-list-item"
        style="cursor: pointer; border: 5px solid #00664f; border-radius: 20px; width: 85%;"
      >
        <div
          class="contract-details"
        >
          <h3
            class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
          >
            Axis Europe (X) PLC
          </h3>
          <p>
            Active contracts: 
            <span
              style="font-weight: 800;"
            />
          </p>
        </div>
        <div
          class="govuk-!-margin-0"
        >
          <span
            class="arrow right"
          />
        </div>
      </li>
      <li
        class="govuk-!-margin-top-3 operative-work-order-list-item"
        style="cursor: pointer; border: 5px solid #00664f; border-radius: 20px; width: 85%;"
      >
        <div
          class="contract-details"
        >
          <h3
            class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
          >
            Axis Europe (X) PLC
          </h3>
          <p>
            Active contracts: 
            <span
              style="font-weight: 800;"
            />
          </p>
        </div>
        <div
          class="govuk-!-margin-0"
        >
          <span
            class="arrow right"
          />
        </div>
      </li>
      <li
        class="govuk-!-margin-top-3 operative-work-order-list-item"
        style="cursor: pointer; border: 5px solid #00664f; border-radius: 20px; width: 85%;"
      >
        <div
          class="contract-details"
        >
          <h3
            class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
          >
            Axis Europe (X) PLC
          </h3>
          <p>
            Active contracts: 
            <span
              style="font-weight: 800;"
            />
          </p>
        </div>
        <div
          class="govuk-!-margin-0"
        >
          <span
            class="arrow right"
          />
        </div>
      </li>
      <li
        class="govuk-!-margin-top-3 operative-work-order-list-item"
        style="cursor: pointer; border: 5px solid #00664f; border-radius: 20px; width: 85%;"
      >
        <div
          class="contract-details"
        >
          <h3
            class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
          >
            Axis Europe (X) PLC
          </h3>
          <p>
            Active contracts: 
            <span
              style="font-weight: 800;"
            />
          </p>
        </div>
        <div
          class="govuk-!-margin-0"
        >
          <span
            class="arrow right"
          />
        </div>
      </li>
      <li
        class="govuk-!-margin-top-3 operative-work-order-list-item"
        style="cursor: pointer; border: 5px solid #00664f; border-radius: 20px; width: 85%;"
      >
        <div
          class="contract-details"
        >
          <h3
            class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
          >
            Herts Heritage Ltd
          </h3>
          <p>
            Active contracts: 
            <span
              style="font-weight: 800;"
            />
          </p>
        </div>
        <div
          class="govuk-!-margin-0"
        >
          <span
            class="arrow right"
          />
        </div>
      </li>
      <li
        class="govuk-!-margin-top-3 operative-work-order-list-item"
        style="cursor: pointer; border: 5px solid #00664f; border-radius: 20px; width: 85%;"
      >
        <div
          class="contract-details"
        >
          <h3
            class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
          >
            Avonline Network (A) Ltd
          </h3>
          <p>
            Active contracts: 
            <span
              style="font-weight: 800;"
            />
          </p>
        </div>
        <div
          class="govuk-!-margin-0"
        >
          <span
            class="arrow right"
          />
        </div>
      </li>
      <li
        class="govuk-!-margin-top-3 operative-work-order-list-item"
        style="cursor: pointer; border: 5px solid #00664f; border-radius: 20px; width: 85%;"
      >
        <div
          class="contract-details"
        >
          <h3
            class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
          >
            Foster Property Maintenance Ltd
          </h3>
          <p>
            Active contracts: 
            <span
              style="font-weight: 800;"
            />
          </p>
        </div>
        <div
          class="govuk-!-margin-0"
        >
          <span
            class="arrow right"
          />
        </div>
      </li>
      <li
        class="govuk-!-margin-top-3 operative-work-order-list-item"
        style="cursor: pointer; border: 5px solid #00664f; border-radius: 20px; width: 85%;"
      >
        <div
          class="contract-details"
        >
          <h3
            class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
          >
            Foster Property Maintenance Ltd
          </h3>
          <p>
            Active contracts: 
            <span
              style="font-weight: 800;"
            />
          </p>
        </div>
        <div
          class="govuk-!-margin-0"
        >
          <span
            class="arrow right"
          />
        </div>
      </li>
      <li
        class="govuk-!-margin-top-3 operative-work-order-list-item"
        style="cursor: pointer; border: 5px solid #00664f; border-radius: 20px; width: 85%;"
      >
        <div
          class="contract-details"
        >
          <h3
            class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
          >
            John Bold & Co LIMITED
          </h3>
          <p>
            Active contracts: 
            <span
              style="font-weight: 800;"
            />
          </p>
        </div>
        <div
          class="govuk-!-margin-0"
        >
          <span
            class="arrow right"
          />
        </div>
      </li>
      <li
        class="govuk-!-margin-top-3 operative-work-order-list-item"
        style="cursor: pointer; border: 5px solid #00664f; border-radius: 20px; width: 85%;"
      >
        <div
          class="contract-details"
        >
          <h3
            class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
          >
            Apex Lifts
          </h3>
          <p>
            Active contracts: 
            <span
              style="font-weight: 800;"
            />
          </p>
        </div>
        <div
          class="govuk-!-margin-0"
        >
          <span
            class="arrow right"
          />
        </div>
      </li>
      <li
        class="govuk-!-margin-top-3 operative-work-order-list-item"
        style="cursor: pointer; border: 5px solid #00664f; border-radius: 20px; width: 85%;"
      >
        <div
          class="contract-details"
        >
          <h3
            class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
          >
            Top Services Ltd
          </h3>
          <p>
            Active contracts: 
            <span
              style="font-weight: 800;"
            />
          </p>
        </div>
        <div
          class="govuk-!-margin-0"
        >
          <span
            class="arrow right"
          />
        </div>
      </li>
      <li
        class="govuk-!-margin-top-3 operative-work-order-list-item"
        style="cursor: pointer; border: 5px solid #00664f; border-radius: 20px; width: 85%;"
      >
        <div
          class="contract-details"
        >
          <h3
            class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
          >
            Sureserve Fire & Electrical
          </h3>
          <p>
            Active contracts: 
            <span
              style="font-weight: 800;"
            />
          </p>
        </div>
        <div
          class="govuk-!-margin-0"
        >
          <span
            class="arrow right"
          />
        </div>
      </li>
      <li
        class="govuk-!-margin-top-3 operative-work-order-list-item"
        style="cursor: pointer; border: 5px solid #00664f; border-radius: 20px; width: 85%;"
      >
        <div
          class="contract-details"
        >
          <h3
            class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
          >
            Aim Windows
          </h3>
          <p>
            Active contracts: 
            <span
              style="font-weight: 800;"
            />
          </p>
        </div>
        <div
          class="govuk-!-margin-0"
        >
          <span
            class="arrow right"
          />
        </div>
      </li>
      <li
        class="govuk-!-margin-top-3 operative-work-order-list-item"
        style="cursor: pointer; border: 5px solid #00664f; border-radius: 20px; width: 85%;"
      >
        <div
          class="contract-details"
        >
          <h3
            class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
          >
            Alphatrack (S) Systems Lt
          </h3>
          <p>
            Active contracts: 
            <span
              style="font-weight: 800;"
            />
          </p>
        </div>
        <div
          class="govuk-!-margin-0"
        >
          <span
            class="arrow right"
          />
        </div>
      </li>
      <li
        class="govuk-!-margin-top-3 operative-work-order-list-item"
        style="cursor: pointer; border: 5px solid #00664f; border-radius: 20px; width: 85%;"
      >
        <div
          class="contract-details"
        >
          <h3
            class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
          >
            Sycous Limited
          </h3>
          <p>
            Active contracts: 
            <span
              style="font-weight: 800;"
            />
          </p>
        </div>
        <div
          class="govuk-!-margin-0"
        >
          <span
            class="arrow right"
          />
        </div>
      </li>
      <li
        class="govuk-!-margin-top-3 operative-work-order-list-item"
        style="cursor: pointer; border: 5px solid #00664f; border-radius: 20px; width: 85%;"
      >
        <div
          class="contract-details"
        >
          <h3
            class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
          >
            THE WIGGETT GROUP LTD
          </h3>
          <p>
            Active contracts: 
            <span
              style="font-weight: 800;"
            />
          </p>
        </div>
        <div
          class="govuk-!-margin-0"
        >
          <span
            class="arrow right"
          />
        </div>
      </li>
      <li
        class="govuk-!-margin-top-3 operative-work-order-list-item"
        style="cursor: pointer; border: 5px solid #00664f; border-radius: 20px; width: 85%;"
      >
        <div
          class="contract-details"
        >
          <h3
            class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
          >
            THE WIGGETT GROUP LTD
          </h3>
          <p>
            Active contracts: 
            <span
              style="font-weight: 800;"
            />
          </p>
        </div>
        <div
          class="govuk-!-margin-0"
        >
          <span
            class="arrow right"
          />
        </div>
      </li>
      <li
        class="govuk-!-margin-top-3 operative-work-order-list-item"
        style="cursor: pointer; border: 5px solid #00664f; border-radius: 20px; width: 85%;"
      >
        <div
          class="contract-details"
        >
          <h3
            class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
          >
            HH General Building Repai
          </h3>
          <p>
            Active contracts: 
            <span
              style="font-weight: 800;"
            />
          </p>
        </div>
        <div
          class="govuk-!-margin-0"
        >
          <span
            class="arrow right"
          />
        </div>
      </li>
      <li
        class="govuk-!-margin-top-3 operative-work-order-list-item"
        style="cursor: pointer; border: 5px solid #00664f; border-radius: 20px; width: 85%;"
      >
        <div
          class="contract-details"
        >
          <h3
            class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
          >
            Conrad Jacobs Ltd
          </h3>
          <p>
            Active contracts: 
            <span
              style="font-weight: 800;"
            />
          </p>
        </div>
        <div
          class="govuk-!-margin-0"
        >
          <span
            class="arrow right"
          />
        </div>
      </li>
    </ol>
  </div>
</DocumentFragment>
`;
